icu4c/source/test/intltest/numbertest_decimalquantity.cpp

```diff
@@ -467,4 +469,191 @@ void DecimalQuantityTest::testNickelRounding() {
+
+        assertEquals(
+                u"formatted number " + cas.skeleton + u" %s toString: " + cas.input,
+                cas.expectedString,
+                actualString);
+        assertDoubleEquals(
+                u"compact decimal " + cas.skeleton + u" n operand: " + cas.input,
+                expectedNOperand,
+                actualNOperand);
+        assertDoubleEquals(
+                u"compact decimal " + cas.skeleton + u" i operand: " + cas.input,
+                expectedIOperand,
+                actualIOperand);
+        assertDoubleEquals(
+                u"compact decimal " + cas.skeleton + " e operand: " + cas.input,
+                expectedEOperand,
+                actualEOperand);
+    }
+}
+
+void DecimalQuantityTest::testSuppressedExponentUnchangedByInitialScaling() {
+    IcuTestErrorCode status(*this, "testCompactDecimalSuppressedExponent");
+    Locale ulocale((const char *)"fr-FR");
+    LocalizedNumberFormatter withLocale = NumberFormatter::withLocale(ulocale);
+    LocalizedNumberFormatter compactLong =
+        withLocale.notation(Notation::compactLong());
+    LocalizedNumberFormatter compactScaled =
+        compactLong.scale(Scale::powerOfTen(3));
+    
+    struct TestCase {
+        int32_t input;
+        UnicodeString expectedString;
+        double expectedNOperand;
+        double expectedIOperand;
+        double expectedEOperand;
+    } cases[] = {
+        // input, compact long string output,
+        // compact n operand, compact i operand, compact e operand
+        {123456789, "123 millions", 123000000.0, 123000000.0, 6.0},
+        {1234567,   "1,2 million",  1200000.0,   1200000.0,   6.0},
+        {123456,    "123 mille",    123000.0,    123000.0,    3.0},
+        {123,       "123",          123.0,       123.0,       0.0},
+    };
+
+    for (const auto& cas : cases) {
+        FormattedNumber fnCompactScaled = compactScaled.formatInt(cas.input, status);
+        DecimalQuantity dqCompactScaled;
+        fnCompactScaled.getDecimalQuantity(dqCompactScaled, status);
+        double compactScaledEOperand = dqCompactScaled.getPluralOperand(PLURAL_OPERAND_E);
+
+        FormattedNumber fnCompact = compactLong.formatInt(cas.input, status);
+        DecimalQuantity dqCompact;
+        fnCompact.getDecimalQuantity(dqCompact, status);
+        UnicodeString actualString = fnCompact.toString(status);
+        double compactNOperand = dqCompact.getPluralOperand(PLURAL_OPERAND_N);
+        double compactIOperand = dqCompact.getPluralOperand(PLURAL_OPERAND_I);
+        double compactEOperand = dqCompact.getPluralOperand(PLURAL_OPERAND_E);
+        assertEquals(
+                u"formatted number compactLong toString: " + Int64ToUnicodeString(cas.input),
+                cas.expectedString,
+                actualString);
+        assertDoubleEquals(
+                u"compact decimal %d, n operand vs. expected" + Int64ToUnicodeString(cas.input),
```

Comment:
Where does the `%d` get substituted?

Reply:
Done.  Split up the test msg string and used string concatenation.

Reply:
If it's a double you can use `DoubleToUnicodeString` instead of `Int64ToUnicodeString`

Reply:
Done.